### ISSUE TITLE
List synonyms on catalog pages

### ISSUE THREAD

**Issue by jonkerz** [calacademy-research/antcat#279]
Related to #275.

**Comment by jonkerz** [comment]
I have deployed code for this on the test instance, please see http://ec2-34-213-24-19.us-west-2.compute.amazonaws.com/catalog/457338

@bpescador Could you see if it's what we want?

---

There's another thing you could help me clear up before pushing to live. It has to do with how we store synonyms in the db. Pretty much: can synonyms be "nested"? I'm not sure how to explain it succinctly so, please bear with me :) Say this happened in the 19th century:

#### 1850s (1 valid species)
- A new species is described: *Antcatia originalium* - Smith, 1850

#### 1860s (3 valid species)
- A new species is described: *Antcatia secondium* - Smith, 1860
- A new species is described: *Antcatia twicesynonymium* - Smith, 1861

#### 1870s (2 valid species, 1 synonym)
- *A. twicesynonymium* is synonymized under *A. secondium*

#### 1880s (1 valid species, 2 synonyms)
- *A. secondium* is synonymized under *A. originalium*

With the above history, there would be a single valid species and two synonyms, and I understand that the other two would be considered to be synonyms of *A. originalium*.

Questions:

- In science, would one also say that *A. twicesynonymium* is a synonym of *A. secondium*?
- If so, does this mean that a synonym can itself have synonyms?
- Does it also mean that a taxon can have both junior and senior synonyms at the same time?
- Does it also mean that a junior synonym can have more than one senior synonyms? Like, would one say that *A. twicesynonymium* has two senior synonyms?
- On AntCat, should we store the synonyms data like this? I'm referring to 'synonyms' as in the 'Synonyms' section on the edit page, so not history items.

I'm asking because we're currently doing it like this on AC, but not 100% consistently, or even 10% consistently. It makes some things complicated under the hood; it's not obvious that this is an issue from just browsing the catalog because we're using some other tricks such as `current_valid_taxon` and  but when I implemented the synonyms list, I realized that some synonyms were missing when I simply listed the synonyms of a taxon without included synonyms of synonyms, or "nested synonyms".

---

Please also see these scripts:

- [Synonyms: Valid taxa listed as another taxons junior synonym](http://ec2-34-213-24-19.us-west-2.compute.amazonaws.com/database_scripts/valid_taxa_listed_as_another_taxons_junior_synonym)
- [Synonyms: Taxa with both junior and senior synonyms](http://ec2-34-213-24-19.us-west-2.compute.amazonaws.com/database_scripts/taxa_with_both_junior_and_senior_synonyms)
- [Synonyms: Junior synonyms listed as another taxons senior](http://ec2-34-213-24-19.us-west-2.compute.amazonaws.com/database_scripts/junior_synonyms_listed_as_another_taxons_senior)

Should these be empty? Should I include these scripts in the next release?

**Comment by bpescador** [comment]
See responses below:

Questions:

   - In science, would one also say that *A. twicesynonymium* is a synonym
   of *A. secondium*?

Yes, it  is is a junior synonym

   - If so, does this mean that a synonym can itself have synonyms



a junior synonym has a senior synonym. S senior synonym has a junior
synonym.   However, I think when we just state what type of "synonym" to
avoid confusion.

   - Does it also mean that a taxon can have both junior and senior
   synonyms at the same time?

Though a single taxon can be a junior synonym and have a senior synonym, it
can not also have a junior synonym.  A senior synonym can have many junior
synonyms.  A junior synonym can have only one senior synonym that is
current.

   - Does it also mean that a junior synonym can have more than one senior
   synonyms? Like, would one say that *A. twicesynonymium* has two senior
   synonyms?

No, a junior synonym can have only one current senior synonym (it may have
more than one listed in the history but only the one with the most recent
date is current).

   - On AntCat, should we store the synonyms data like this? I'm referring
   to 'synonyms' as in the 'Synonyms' section on the edit page, so not history
   items.


So to summarize the rules (very similar to your scripts which should return
zero (so much work to do)

*A valid taxon cannot have another taxon as a senior synonym, which means a
valid taxon cannot be a junior synonym of any other taxon.*
*A taxon  cannot have both junior and senior synonyms.*
*A taxon that is a junior synonym cannot be another taxon's senior synonym
(just valid taxa can be senior synonym).*

We should count the number of junior synonyms for valid taxa only.  So in
this example:
http://ec2-34-213-24-19.us-west-2.compute.amazonaws.com/catalog/457338

I think we should state:  7 junior synonyms.

We only need to report junior synonyms for the valid taxa.


On Wed, Sep 27, 2017 at 12:13 PM, Fredrik Palmkron <notifications@github.com

**Comment by jonkerz** [comment]
Thanks for all the answers!

> I think we should state:  7 junior synonyms.

I've changed the subheading.

> We only need to report junior synonyms for the valid taxa.

The subheading will not be shown when there are not junior synonyms, but I decided include it regardless of whether a taxon is valid or not, because if I understood all your answers correctly, this will sort itself out once the taxa in the scripts have been fixed.

**Comment by bpescador** [comment]
Sounds good!
